### PR TITLE
Feature: Add Python 3.9 support

### PR DESCRIPTION
**Pull request by BoboTiG**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -83,6 +83,11 @@ jobs:
     - <<: *test-libraries
       python: 3.8
 
+    - <<: *test-pyinstaller
+      python: 3.9-dev
+    - <<: *test-libraries
+      python: 3.9-dev
```
Comment (Legorooj) [inline .travis.yml:89]: 3.9 is officially released now, you should be able to just do `3.9`?

Comment (BoboTiG) [reply .travis.yml:89]: You are right, just in time!

File: tests/requirements-libraries.txt [hunk 2]
```diff
@@ -18,7 +18,7 @@
 # - v. 2.2 and above fails.
 Django==2.1.8  # pyup: ignore
 matplotlib==3.0.3; python_version <= '3.5'  # pyup: ignore
-matplotlib==3.2.2; python_version >= '3.6'  # pyup: ignore
+matplotlib==3.2.2; python_version >= '3.6' and python_version < '3.9'  # pyup: ignore
```
Comment (Legorooj) [inline tests/requirements-libraries.txt:21]: Shouldn't this be in `Python 3.9 not yet supported`?

File: news/5289.feature.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Added Python 3.9 support.
```
Comment (htgoebel) [inline news/5289.feature.rst:1]: Please follow our guideline for commit messages, esp. use prsent tense.

File: tests/requirements-libraries.txt [hunk 4]
```diff
@@ -17,8 +17,6 @@
 #
 # - v. 2.2 and above fails.
 Django==2.1.8  # pyup: ignore
-matplotlib==3.0.3; python_version <= '3.5'  # pyup: ignore
-matplotlib==3.2.2; python_version >= '3.6'  # pyup: ignore
```
Comment (htgoebel) [inline tests/requirements-libraries.txt:21]: This will complety remove testing matplot. How is this related to Python 3.9?

File: tests/requirements-libraries.txt [hunk 5]
```diff
@@ -30,7 +28,9 @@ babel==2.8.0
 future==0.18.2
 gevent==20.9.0
 pygments==2.7.1
-PySide2==5.14.0
+pyside2==5.15.1
+pyqt5==5.15.1
+pyqtwebengine==5.15.1
```
Comment (htgoebel) [inline tests/requirements-libraries.txt:33]: How is this realted to Python 3.0?

File: tests/requirements-libraries.txt [hunk 6]
```diff
@@ -51,28 +52,23 @@ ipython==7.16.1; python_version == '3.6'  # pyup: ignore
 ipython==7.9.0; python_version == '3.5'  # pyup: ignore
 
 # pandas 1.0.0 dropped Python 3.5 support.
-pandas==1.1.3; python_version > '3.5'
+# Not yet Python 3.9 compatible because of Numpy
+pandas==1.1.3; python_version > '3.5' and python_version < '3.9'
```
Comment (htgoebel) [inline tests/requirements-libraries.txt:56]: I understand that numpy ist not yet available for 3.9- But this change inhibits pandas to be tested again a soon ans it is available for Python 3.9. So we need to what pandas/numpy for py3.9 support. Anyhow, this should not be necessay, since if there is no pandas for 3.9, tests are skiped.

File: tests/requirements-libraries.txt [hunk 7]
```diff
@@ -51,28 +52,23 @@ ipython==7.16.1; python_version == '3.6'  # pyup: ignore
...
 pandas==0.25.3; python_version == '3.5'  # pyup: ignore
 
 # Numpy 1.19 dropped Python 3.5 support
-numpy==1.19.2; python_version > '3.5'
+numpy==1.19.2; python_version > '3.5' and python_version < '3.9'
```
Comment (htgoebel) [inline tests/requirements-libraries.txt:60]: See commante for pandas above.

File: tests/requirements-libraries.txt [hunk 8]
```diff
@@ -51,28 +52,23 @@ ipython==7.16.1; python_version == '3.6'  # pyup: ignore
...
 numpy<=1.19.0; python_version == '3.5'  # pyup: ignore
 
 # And so did SciPy
-scipy==1.5.3; python_version > '3.5'
+# Not yet Python 3.9 compatible because of Numpy
+scipy==1.5.3; python_version > '3.5' and python_version < '3.9'
```
Comment (htgoebel) [inline tests/requirements-libraries.txt:65]: See commante for pandas above.

File: tests/requirements-libraries.txt [hunk 9]
```diff
@@ -51,28 +52,23 @@ ipython==7.16.1; python_version == '3.6'  # pyup: ignore
...
-# Special cases
-# -------------
-
-# Per https://www.riverbankcomputing.com/pipermail/pyqt/2020-February/042506.html, Windows PyQt5 5.14.0? and 5.14.1? aren't correctly packaged.
-pyqt5==5.13.2; sys_platform == 'win32'
-pyqt5==5.14.1; sys_platform != 'win32'
-# On Appveyor, newer versions fail with the error message
-# ``[1916:2732:0204/144922.809:ERROR:mf_helpers.cc(14)] Error in dxva_video_decode_accelerator_win.cc on line 513``
-pyqtwebengine==5.12.1; sys_platform == 'win32'  # pyup: ignore
-pyqtwebengine==5.14.0; sys_platform != 'win32'  # pyup: ignore
```
Comment (htgoebel) [inline tests/requirements-libraries.txt:78]: Wha have all these spcaial cases been removed?

File: tests/requirements-libraries.txt [hunk 10]
```diff
@@ -51,28 +52,23 @@ ipython==7.16.1; python_version == '3.6'  # pyup: ignore
...
+# Not yet Python 3.9 compatible because of Numpy
+matplotlib==3.0.3; python_version <= '3.5'  # pyup: ignore
+matplotlib==3.2.2; python_version >= '3.6' and python_version < '3.9'  # pyup: ignore
```
Comment (htgoebel) [inline tests/requirements-libraries.txt:74]: Please keep at the old place. Also see comment regarding pandas above.

**Review by htgoebel — CHANGES_REQUESTED** [review]
Thanks for picking this up. Anyhow I have quite some comments:

1. Obviously (as you commented out parts of the test suite) this is a draft. Please run test on you own branch and own Travsi/Appvayer account. This will speed up your testing as the main pipe gets stuck quickly.
2. Please split this up into several commits. This also allows to reason you changes. Please have a look at the commit from 9f8e9a2ab2a946d7c90d78756610cfa125b38f87 to e61fad42cfedd2dbf1d183a39f40355e2360c69a. Thanks.

File: tests/requirements-libraries.txt (hunk 9 above)
Comment (BoboTiG) [reply tests/requirements-libraries.txt:78]: Tests are passing on Windows, so yes :)
I am still working to make this OK on GNU/Linux though (draft PR).

File: tests/requirements-libraries.txt (hunk 4 above)
Comment (BoboTiG) [reply tests/requirements-libraries.txt:21]: matplot (and pandy and scipy) requires numpy. But Numpy is not yet ready for Python 3.9.

BUT I just saw Numpy 1.19.3 is out and it works on Python 3.9. I will update everything.

File: tests/requirements-libraries.txt (hunk 5 above)
Comment (BoboTiG) [reply tests/requirements-libraries.txt:33]: I am not sure I understand your comment 🤔

**Comment by BoboTiG** [comment]
I will reopen when ready (running Travis on my own repos).

File: tests/requirements-libraries.txt (hunk 5 above)
Comment (Legorooj) [reply tests/requirements-libraries.txt:35]: This Qt stuff is fine - it's due to test failures which I'm pretty sure no longer are there in newer versions of Qt.

File: tests/requirements-libraries.txt (hunk 9 above)
Comment (Legorooj) [reply tests/requirements-libraries.txt:78]: The Qt stuff _should_ be fine if you upgrade the versions, but check.
